kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -110,7 +114,16 @@ public ActionWrapper next() {
             throw new NoSuchElementException("No next element");
         }
 
-        return actionsIter.get().next();
+        ActionWrapper next = actionsIter.get().next();
+
+        // If the ActionWrapper schema does not match the read schema, then we have injected sidecar
+        // files into the schema. Append any sidecar files found to the end of the file list and
+        // return a new ActionWrapper with the sidecar files removed from the read schema.
+        if (!next.getColumnarBatch().getSchema().equals(readSchema) && next.isFromCheckpoint()) {
```

Comment:
Instead of comparing the schema, can we set the state in `ActionsIterator` to indicate the current data is being read from a v2 checkpoint manifest file?

Reply:
Done.